src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/StdLibRepresentationTests.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.checkImpl
+import org.jetbrains.kotlinx.lincheck.strategy.managed.forClasses
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.junit.Test
+import java.util.concurrent.ConcurrentHashMap
+
+class ThreadSafeCollectionMutedRepresentationTest: BaseTraceRepresentationTest("thread_safe_collection_muted") {
+    
+    private val concurrentMap = ConcurrentHashMap<Int, Int>()
+    private var a = 1
+    
+    override fun operation() {
+        concurrentMap.put(1, 1)
+        a++
+    }
+}
+
+class ThreadSafeCollectionMutedLambdaRepresentationTest: BaseTraceRepresentationTest("thread_safe_collection_muted_lambda") {
+
+    private val concurrentMap = ConcurrentHashMap<Int, Int>()
+    private val innerConcurrentMap = ConcurrentHashMap<Int, Int>()
+    private var a = 1
+
+    override fun operation() {
+        concurrentMap.computeIfAbsent(1) { 
+            a = 5
+            innerConcurrentMap.computeIfAbsent(1) { ++a }
+            ++a 
+        }
+        a++
+    }
+}
+
+class ThreadSafeCollectionUnmutedRepresentationTest: BaseTraceRepresentationTest("thread_safe_collection_unmuted") {
```

Comment:
Please move this test to appear right after `ThreadSafeCollectionMutedRepresentationTest`